indexing-service/src/main/java/org/apache/druid/indexing/common/task/BatchAppenderators.java

```diff
@@ -59,7 +65,9 @@ public static Appenderator newAppenderator(
       TaskToolbox toolbox,
       DataSchema dataSchema,
       AppenderatorConfig appenderatorConfig,
-      DataSegmentPusher segmentPusher
+      DataSegmentPusher segmentPusher,
+      RowIngestionMeters rowIngestionMeters,
+      ParseExceptionHandler parseExceptionHandler
```

Comment:
I see your point, but I'm not sure if it's good to have `parseExceptionHandler` be a `rowIngestionMeters` provider. I guess we have a couple of options for alternatives:

- Merge `RowIngestioMeters` and `ParseExceptionHandler`. `RowIngestionMeters` will have `handle(ParseException)` method and throw an exception when it hits the limit. This seems a bit strange to me since metrics registry can throw an exception.
- Make `RowIngestionMeters` to be only a registry and use individual `Meter`s. For example, `ParseExceptionHandler` will have only a `Meter` for counting unparseable events. `AppenderatorImpl` will have only a `Meter` for counting processed events. This seems sane, but can make code error-prone since we will have multiple `Meter`s which should be properly chosen whenever we use them.

I'm not sure any of these are better than now. Do you have any better idea?

Reply:
fwiw, I think current implementation is better one.